grpc/shared/rollupdata_handling.go

```diff
@@ -0,0 +1,261 @@
+	}
+
+	if deposit.Asset != bac.AssetDenom {
+		return nil, fmt.Errorf("asset %s does not match bridge address %s asset", deposit.Asset, bridgeAddress)
+	}
+
+	recipient := common.HexToAddress(deposit.DestinationChainAddress)
+	amount := bac.ScaledDepositAmount(protoU128ToBigInt(deposit.Amount))
+
+	if bac.Erc20Asset != nil {
+		log.Debug("creating deposit tx to mint ERC20 asset", "token", bac.AssetDenom, "erc20Address", bac.Erc20Asset.ContractAddress)
+		abi, err := contracts.AstriaBridgeableERC20MetaData.GetAbi()
+		if err != nil {
+			// this should never happen, as the abi is hardcoded in the contract bindings
+			return nil, fmt.Errorf("failed to get abi for erc20 contract for asset %s: %w", bac.AssetDenom, err)
+		}
+
+		// pack arguments for calling the `mint` function on the ERC20 contract
+		args := []interface{}{recipient, amount}
+		calldata, err := abi.Pack("mint", args...)
+		if err != nil {
+			return nil, err
+		}
+
+		txdata := types.DepositTx{
+			From:  bac.SenderAddress,
+			Value: new(big.Int), // don't need to set this, as we aren't minting the native asset
+			// mints cost ~14k gas, however this can vary based on existing storage, so we add a little extra as buffer.
+			//
+			// the fees are spent from the "bridge account" which is not actually a real account, but is instead some
+			// address defined by consensus, so the gas cost is not actually deducted from any account.
+			Gas:                    64000,
+			To:                     &bac.Erc20Asset.ContractAddress,
+			Data:                   calldata,
+			SourceTransactionId:    *deposit.SourceTransactionId,
+			SourceTransactionIndex: deposit.SourceActionIndex,
+		}
+
+		tx := types.NewTx(&txdata)
+		return tx, nil
+	}
+
+	txdata := types.DepositTx{
+		From:                   bac.SenderAddress,
+		To:                     &recipient,
+		Value:                  amount,
+		Gas:                    0,
+		SourceTransactionId:    *deposit.SourceTransactionId,
+		SourceTransactionIndex: deposit.SourceActionIndex,
+	}
+	return types.NewTx(&txdata), nil
+}
+
+func validateAndUnmarshallSequenceAction(tx *sequencerblockv1.RollupData) (*types.Transaction, error) {
+	ethTx := new(types.Transaction)
+	err := ethTx.UnmarshalBinary(tx.GetSequencedData())
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal sequenced data into transaction: %w. tx hash: %s", err, sha256.Sum256(tx.GetSequencedData()))
+	}
+
+	if ethTx.Type() == types.DepositTxType {
+		return nil, fmt.Errorf("deposit tx not allowed in sequenced data. tx hash: %s", sha256.Sum256(tx.GetSequencedData()))
+	}
+
+	if ethTx.Type() == types.BlobTxType {
+		return nil, fmt.Errorf("blob tx not allowed in sequenced data. tx hash: %s", sha256.Sum256(tx.GetSequencedData()))
+	}
+
+	return ethTx, nil
+}
+
+func unmarshallAllocationTxs(allocation *auctionv1alpha1.Allocation, prevBlockHash []byte, auctioneerBech32Address string, addressPrefix string) (types.Transactions, error) {
```

Comment:
```suggestion
func unmarshalAllocationTxs(allocation *auctionv1alpha1.Allocation, prevBlockHash []byte, auctioneerBech32Address string, addressPrefix string) (types.Transactions, error) {
```

Reply:
addressed in https://github.com/astriaorg/flame/pull/30/commits/defc35bb0c32331a823a635dccdb28c7e62d519b